HeterogeneousCore/CUDAUtilities/interface/CUDAEventCache.h

```diff
@@ -1,30 +1,43 @@
 #ifndef HeterogeneousCore_CUDAUtilities_CUDAEventCache_h
 #define HeterogeneousCore_CUDAUtilities_CUDAEventCache_h
 
-#include <memory>
+#include <vector>
 
-#include <cuda/api_wrappers.h>
+#include <cuda_runtime.h>
 
 #include "FWCore/Utilities/interface/ReusableObjectHolder.h"
+#include "HeterogeneousCore/CUDAUtilities/interface/SharedEventPtr.h"
 
 class CUDAService;
 
 namespace cudautils {
   class CUDAEventCache {
   public:
+    using BareEvent = SharedEventPtr::element_type;
```

Comment:
is this a plain `cudaEvent_t` ?

Reply:
It is `CUevent` (pointed-to type of `cudaEvent_t`), but I'd rather not explicitly use types from the driver API.

Reply:
ah ok - yes, agreed